### PR TITLE
Introduce fine tuning for storm topologies

### PR DESCRIPTION
**Pull request by sergii-iakovenko**
The changes:
- Introduce per topology config file, utilize Storm Flux for it.
- Generate topology definitions by confd templates.

### CODE REVIEW

**Review by dpoltavets — APPROVED** [review]

File: src-java/connecteddevices-topology/connecteddevices-storm-topology/src/main/resources/connecteddevices-topology.yaml.example [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# topology configuration
```
Comment (surabujin) [inline src-java/connecteddevices-topology/connecteddevices-storm-topology/src/main/resources/connecteddevices-topology.yaml.example:1]: I would like to not have these *-topology.yaml.exmapl files. There are too many of them. We have issues with supporting ... much lover count of example config for existing components. We will fail to support these examples.

File: src-java/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.1.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.7-bin.zip
```
Comment (surabujin) [inline src-java/gradle/wrapper/gradle-wrapper.properties:3]: Is it an intentional change?

**Review by surabujin — APPROVED** [review]

**Review by niksv — APPROVED** [review]

**Review by timofei-durakov — APPROVED** [review]

File: src-java/connecteddevices-topology/connecteddevices-storm-topology/src/main/resources/connecteddevices-topology.yaml.example (hunk 1 above)
Comment (sergii-iakovenko) [reply src-java/connecteddevices-topology/connecteddevices-storm-topology/src/main/resources/connecteddevices-topology.yaml.example:1]: Make sense. Removed.

File: src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/topology/AbstractTopology.java [hunk 3]
```diff
@@ -99,12 +111,36 @@ protected AbstractTopology(LaunchEnvironment env, Class<T> topologyConfigClass)
...
+                topologyName, kafkaConfig.getHosts(), getTopologyParallelism(), getTopologyWorkers());
     }
 
-    protected String getDefaultTopologyName() {
-        return getClass().getSimpleName().toLowerCase();
+    private Optional<TopologyDef> loadTopologyDef(String topologyDefinitionName, Properties properties) {
+        try {
+            // Check the definition in resources for existence.
+            String yamlResource = format("/%s.yaml", topologyDefinitionName);
+            if (FluxParser.class.getResourceAsStream(yamlResource) == null) {
```
[the remark this answers is not in the document]
Comment (sergii-iakovenko) [reply src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/topology/AbstractTopology.java:127]: For topology definition files I reuse the templating approach we apply for topology.properties. So a developer defines which Storm components require individual configuration, and puts it into the corresponding topology definition. Then during the deployment, we can override the file in the way we do for topology.properties.

File: src-java/gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (sergii-iakovenko) [reply src-java/gradle/wrapper/gradle-wrapper.properties]: Nope. Reverted.

**Comment by sergii-iakovenko** [comment]
Related to #3051

**Pull request merged**